airflow/api_fastapi/openapi/v1-generated.yaml

```diff
@@ -155,7 +155,7 @@ paths:
         required: false
         schema:
           type: string
-          default: dag_id
+          default: ''
```

Comment:
We lost the default here because the value is now dinamically set at runtime by inspecting the model

Reply:
Sure, I am checking the possibilities and will propose something soon. Agree, this is not looking good. Thanks!

Reply:
I have included a solution that works fine for only SortParam cases. I'm unsure if we want to generalise the approach for all dynamic default values for method parameters. In that case, we may need a more sophisticated parsing mechanism for method signatures. I think this can work until we have another exceptional case. If that is so, we can see better how to approach multiple use cases of updating the openapi specs. 

I have checked if we can somehow include additional parameters to APIRoute via injecting similar to AirflowRouter implementation. It needs more than overriding a method from the APIRouter class ([add_api_route](https://github.com/fastapi/fastapi/blob/master/fastapi/routing.py#L881)) and including a custom APIRoute class. So I am not sure if we should spend more time fitting a dynamic solution to include this information to the FastAPI objects which would be then cascaded to openapi spec in the implementation.

What do you think?

Reply:
I don't really like the introspection / custom code to regenerate the defaults spec values. All that seems easy to break and a lot of boilerplate code to maintain.

What I had in mind was more to find a way to make FastAPI detect the default correctly. I'll take a look and see if I can find another approach.

Reply:
Yeah, I agree. I can not say I liked the solution either.

I will also spend a bit more time to fit it in FastAPI. 
Even though I tried overriding some of the classes and their methods of FastAPI, somehow, it still tries to call the base implementation. Most prob, I missed something along the way.

Reply:
I just pushed a new commit on the branch. It removes the need for custom code to populate the documentation.

Let me know what you think.

https://github.com/apache/airflow/pull/42782/commits/9c056a75f99f6a8a30a6493012a4881204d805ba

Reply:
I could only find time to check this out. I did a bit of complex thinking there. Thanks a lot for the changes! It looks great! `Callable` inception :100: